### PR TITLE
Remove old malloc trace implementation

### LINKED ISSUE #1865: Remove old malloc trace implementation
Remove the old malloc trace implementation, since we have a new one and the old one doesn't work on newer glibc versions.

### PR DESCRIPTION
**Pull request by schmelter-sap**
Removes the old malloc trace implementation.

fixes #1865

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

File: src/java.base/share/native/libjli/java.c [hunk 1]
```diff
@@ -962,7 +955,7 @@ jboolean ShouldPreloadLibMallocHooks(int argc, char **argv) {
         }
     }
 
-    return (uses_new_trace && !uses_old_trace) ? JNI_TRUE : JNI_FALSE;
+    return uses_new_trace ? JNI_TRUE : JNI_FALSE;
```
Comment (reinrich) [inline src/java.base/share/native/libjli/java.c]: Just return `uses_new_trace`.
And better rename it. Maybe `trace_is_enabled`?

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Review by reinrich — APPROVED** [review]
Looks good 👍

**Review by RealCLanger — APPROVED** [review]
Richards findings were what I found as well. Tests are good, failures are unrelated. So this is good. 👍

**Pull request merged**